packages/react-reconciler/src/ReactProfileTimer.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Fiber} from './ReactFiber';
+import type {StackCursor} from './ReactFiberStack';
+
+import warning from 'fbjs/lib/warning';
+import ReactFiberStack from './ReactFiberStack';
+
+const hasNativePerformanceNow =
+  typeof performance === 'object' && typeof performance.now === 'function';
+
+let now;
+if (hasNativePerformanceNow) {
+  now = function() {
+    return performance.now();
+  };
+} else {
+  now = function() {
+    return Date.now();
+  };
+}
+
+/**
+ * The "actual" render time is total time required to render the descendants of a Profiler component.
+ * This time is stored as a stack, since Profilers can be nested.
+ * This time is started during the "begin" phase and stopped during the "complete" phase.
+ * It is paused (and accumulated) in the event of an interruption or an aborted render.
+ */
+
+const {checkThatStackIsEmpty, createCursor, push, pop} = ReactFiberStack();
+
+let stackCursor: StackCursor<Fiber> = createCursor(null);
```

Comment:
This seems like very odd types. I'm not sure how Flow lets you do this.

You're passing a default argument for `null` which is not compatible with the type `Fiber` which you're assigning to the StackCursor.

Then later you actually pass a `number` to it. So it's three way inconsistent.

I think it's not noticed because you never actually use the values stored in this stack.